### ISSUE TITLE
Sync OBU improvement suggestions

### ISSUE THREAD

**Issue by cconcolato** [AOMediaCodec/iamf#401]
- `global_offset`
  - the definition does not provide a unit. 
  - if it is defined as 0 in this version of the specification, should it really be defined as is? Consider renaming it `reserved` or removing it from the version 0 of the specification.  
  
- `obu_id`: 
  - It should rather say that the value of this field is to be interpreted depending on the value of `obu_type`. If `obu_type` is 0, the `obu_id` is interpreted as a `audio_substream_id`. If `obu_type` is 1, the `obu_id` is interpreted as a `parameter_id`. 
  - I would simply rename it to `id` and also rename `num_obu_ids` simply to `count`
 
- `relative_offset`: 
  - the semantics is generally hard to follow. Consider simply deferring to section 5.2
  - In the sentence "The offset shall be indicated in the number of ticks at the sample rate specified in the corresponding substream or Codec Config OBU.", the "or" word seems wrong.

**Comment by cconcolato** [comment]
I see that the Sync OBU is not used in the 2 profiles currently being defined. If that says like that, why not making it a 'reserved' OBU and leave it for a v2 of the spec?

**Comment by sunghee-hwang** [comment]
- `global_offset`
  - I agree with `reserved` and shall be set to 0.
- `obu_id`
  - Sounds good to me
- `relative_offset`
  - This may be better and more clear. "The offset for `audio_substream_id` shall be indicated in the number of ticks at the sample rate of its associated substream and the offset for `parameter_id` shall be indicated in the number of ticks at the sample rate of the substream which its associated parameter is applied to.
